python/ray/train/lightning/lightning_checkpoint.py

```diff
@@ -0,0 +1,97 @@
+import os
+import logging
+import pytorch_lightning as pl
+import tempfile
+import shutil
+
+from inspect import isclass
+from typing import Optional, Type
+
+from ray.air.constants import MODEL_KEY
+from ray.air._internal.checkpointing import save_preprocessor_to_dir
+from ray.data import Preprocessor
+from ray.train.torch import TorchCheckpoint
+from ray.util.annotations import PublicAPI
+
+logger = logging.getLogger(__name__)
+
+
+@PublicAPI(stability="alpha")
+class LightningCheckpoint(TorchCheckpoint):
+    """A :class:`~ray.air.checkpoint.Checkpoint` with Lightning-specific functionality.
+
+    LightningCheckpoint only support file based checkpoint loading.
+    Create this by calling ``LightningCheckpoint.from_directory(ckpt_dir)``,
+    ``LightningCheckpoint.from_uri(uri)`` or ``LightningCheckpoint.from_path(path)``
+
+    LightningCheckpoint loads file named ``model`` under the specified directory.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._cache_dir = None
+
+    @classmethod
+    def from_path(
+        cls,
+        path: str,
+        *,
+        preprocessor: Optional["Preprocessor"] = None,
+    ) -> "LightningCheckpoint":
+        """Create a ``ray.air.lightning.LightningCheckpoint`` from a checkpoint path.
+        Args:
+            path: The file path to the PyTorch Lightning checkpoint.
+            preprocessor: A fitted preprocessor to be applied before inference.
+        Returns:
+            An :py:class:`LightningCheckpoint` containing the model.
+        Examples:
+            >>> from ray.train.lightning import LightningCheckpoint
+            >>>
+            >>> checkpoint = LightningCheckpoint.from_path("/path/to/checkpoint.ckpt")
+        """
+
+        assert os.path.exists(path), f"Lightning checkpoint {path} doesn't exists!"
+
+        cache_dir = tempfile.mkdtemp()
+        new_checkpoint_path = os.path.join(cache_dir, MODEL_KEY)
+        shutil.copy(path, new_checkpoint_path)
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, cache_dir)
+        checkpoint = cls.from_directory(cache_dir)
+        checkpoint._cache_dir = cache_dir
```

Comment:
do we need this line?

Reply:
We used `tempfile.mkdtemp()` and have to clear this folder afterwards. This cache_dir is deleted in `__del__()` method.